napari/_qt/widgets/magic_gui_dask_settings.py

```diff
@@ -0,0 +1,18 @@
+from magicgui import magicgui
+
+from .magic_gui_base_widget import BaseMagicSetting
+
+
+@magicgui(auto_call=True, layout='horizontal', cache={'min': 0, 'max': 20})
+def dask_settings(dask_enabled=True, cache=15.0) -> int:
+    """Create magic gui function GUI with checkbox and spinbox for dask settings."""
+    return {'enabled': dask_enabled, 'cache': cache}
+
+
+class DaskSettings(BaseMagicSetting):
+    """Class for use in json schema widget builder for dask settings."""
+
+    def __init__(self):
+
+        self.MAGIC_GUI = dask_settings
+        super().__init__()
```

Comment:
Could this code 

```python
class DaskSettings(BaseMagicSetting):
    """Class for use in json schema widget builder for dask settings."""

    def __init__(self):

        self.MAGIC_GUI = dask_settings
        super().__init__()
```

Become

```python
class DaskSettings(BaseMagicSetting):
    """Class for use in json schema widget builder for dask settings."""

    MAGIC_GUI = dask_settings
```

Reply:
@goanpeca I tried your code, and it gives me a `RecursionError`.  The only way I can see to avoid the error is to do what I have above.